src/DroneSafetyApi/Controllers/HeatmapsController.cs

```diff
@@ -27,11 +32,18 @@ public IActionResult Get([FromQuery]HeatmapsQuery query)
             {
                 return new BadRequestResult();
             }
-            // TODO: Decide optimal radius value, and put value/computation in appropriate place.
-            var intersectionHazards = DataPoints.GetDataPointsInRadius(query.Centre, 100000);
-            var heatmapsResponse = DataPointsToHeatmaps.ConvertToHeatmapResponse(
-                query.Resolution,
+            var radius = (int) (Math.Sqrt(((query.Area.Max.Latitude - query.Area.Min.Latitude) *
+                (query.Area.Max.Latitude - query.Area.Min.Latitude)) +
+                ((query.Area.Max.Longitude - query.Area.Min.Longitude) *
+                (query.Area.Max.Longitude - query.Area.Min.Longitude)))) * MetresInLatDegree;
+            var intersectionHazards = ((IEnumerable<Hazard>)
+                Hazards.GetHazardsInRadius<CircularHazard>(query.Centre, radius, "Circle", query.Time))
```

Comment:
The name of the shape shouldn't be passed as a string. There should be separate methods to query each. Also, the concatenation should happen somewhere else (in the repository instance, say). In fact, we should be saving different shapes in different documents, or differentiating between them at the query level as opposed to having to do it with string comparisons.